Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/EOAWalletToEmbeddedWalletAdapter.cs

```diff
@@ -0,0 +1,252 @@
+                                throw new Exception("No Receipt");
+                            }
+                            else
+                            {
+                                var successfullReturn = new SuccessfulTransactionReturn(responseReceipt.transactionHash, null, null, null, JObject.FromObject(responseReceipt));
+                                successfulTransactionReturns.Add(successfullReturn);
+                                OnSendTransactionComplete?.Invoke(successfullReturn);
+                            }
+                        }
+                        catch (Exception receiptEx)
+                        {
+                            Debug.LogError("Error while waiting for the receipt: " + receiptEx.Message);
+                            string receiptExString = receiptEx.Message;
+                            var failedReturn = new FailedTransactionReturn(receiptEx.Message, null, null);
+                            failedTransactionReturns.Add(failedReturn);
+                            OnSendTransactionFailed?.Invoke(failedReturn);
+                        }
+                    }
+                    else
+                    {
+                        var responseReceipt = await Wallet.SendTransaction(client, transaction);
+                        successfulTransactionReturns.Add(new SuccessfulTransactionReturn(responseReceipt, null, null, null));
+                    }
+                }
+            }
+
+            if (failedTransactionReturns.Count == 0)
+            {
+                return successfulTransactionReturns.Count > 1
+                    ? new SuccessfulBatchTransactionReturn(successfulTransactionReturns.ToArray())
+                    : successfulTransactionReturns[0];
+            }
+
+            return failedTransactionReturns.Count > 1
+                ? new FailedBatchTransactionReturn(successfulTransactionReturns.ToArray(), failedTransactionReturns.ToArray())
+                : failedTransactionReturns[0];
+        }
+        
+        public event Action<SuccessfulContractDeploymentReturn> OnDeployContractComplete;
+        public event Action<FailedContractDeploymentReturn> OnDeployContractFailed;
+
+        public async Task<ContractDeploymentReturn> DeployContract(Chain network, string bytecode, string value)
+        {
+            var client = new SequenceEthClient(network);
+
+
+            ContractDeploymentReturn creturn;
+            try
+            {
+                ContractDeploymentResult result = await ContractDeployer.Deploy(client, Wallet, bytecode);
+                if (result.DeployedContractAddress != null)
+                {
+                    creturn = new SuccessfulContractDeploymentReturn(new SuccessfulTransactionReturn(result.Receipt.transactionHash, null, null, null), result.DeployedContractAddress);
+                }
+                else
+                {
+                    throw new Exception(result.TransactionHash);
```

Comment:
Please provide more information in this exception. Something like "Unable to retrieve the deployed contract address from your transaction receipt" provides a lot more information than a transaction hash (which may just look like a random hex string to an integrator)